source/extensions/transport_sockets/tls/context_impl.h

```diff
@@ -160,6 +163,11 @@ class ContextImpl : public virtual Envoy::Ssl::Context {
   std::string cert_chain_file_path_;
   TimeSource& time_source_;
   const unsigned tls_max_version_;
+  mutable Stats::StatNameSet stat_name_set_;
```

Comment:
If I understand correctly, this must be `mutable`, because `getStatName` is non-`const` but called from a `const` method (`incCounter`). Looking at [`getStatName` definition](https://github.com/envoyproxy/envoy/blob/c3a75316f2fa495fc7be36efd4f291445ac7b857/source/common/stats/symbol_table_impl.cc#L445), we probably could make it `const` by making the lock member `mutable` and switching `dynamic_stat_names_` to use `.at(token)` which is `const` instead of `operator[]`. Does this sound right?

Reply:
I don't think so...the use of [] intentionally does an insertion into the map if it does not exist. So if we don't make StatNameSet member-var mutable here, we'd have to make the dynamic map within the StatNameSet be mutable. If we call 'at()' for this use-case it will throw.

Now, we could argue whether getStatName() conceptually mutates the StatNameSet but I think it does, and the doc for getStatName() should indicate more clearly so. I'm adding to the PR a rephrasing in the doc for getStatName().

We could *also* argue whether getStatName() is an appropriate name for a mutating method, and I'd be happy to change it if you have a better idea :)

Reply:
I think the doc addition suffices. Thanks for the explanation.